News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/NewsReaderDetailFragment.java

```diff
@@ -413,18 +421,47 @@ public void onInflate(Context context, AttributeSet attrs, Bundle savedInstanceS
 
         ((NewsReaderApplication) getActivity().getApplication()).getAppComponent().injectFragment(this);
 
-        TypedArray a = context.obtainStyledAttributes(attrs, new int[]{R.attr.markasreadDrawable, R.attr.starredDrawable, R.attr.colorAccent});
-        markAsReadDrawable = a.getDrawable(0);
-        starredDrawable = a.getDrawable(1);
+        TypedArray a = context.obtainStyledAttributes(attrs, new int[]{R.attr.colorAccent});
+        updateSwipeDrawables(true);
         int color = Constants.isNextCloud(mPrefs) ? R.color.nextcloudBlue : R.color.owncloudBlue;
         accentColor = a.getColor(2, ContextCompat.getColor(context, color));
         a.recycle();
     }
 
+    private String prevLeftAction = "";
+    private String prevRightAction = "";
+    private void updateSwipeDrawables(boolean always) {
+        String leftAction  = mPrefs.getString(SP_SWIPE_LEFT_ACTION, SP_SWIPE_LEFT_ACTION_DEFAULT);
+        String rightAction = mPrefs.getString(SP_SWIPE_RIGHT_ACTION, SP_SWIPE_RIGHT_ACTION_DEFAULT);
+        if (!always && leftAction.equals(prevLeftAction) && rightAction.equals(prevRightAction))
+            return;
+        prevLeftAction  = leftAction;
+        prevRightAction = rightAction;
+        int leftId  = getLayoutId(leftAction);
+        int rightId = getLayoutId(rightAction);
+        TypedArray a = getContext().obtainStyledAttributes(new int[]{leftId, rightId});
+        leftSwipeDrawable = a.getDrawable(0);
+        rightSwipeDrawable = a.getDrawable(1);
+    }
+
+    private int getLayoutId(String action) {
+        switch (action) {
+            case "0": return R.attr.openinbrowserDrawable;
+            case "1": return R.attr.starredDrawable;
+            case "2": return R.attr.markasreadDrawable;
+            default:
+                Log.e(TAG, "Invalid option saved to prefs. This should not happen");
+                return Integer.MAX_VALUE;
+        }
+    }
+
     @Override
     public void onViewStateRestored(Bundle savedInstanceState) {
-        if (savedInstanceState != null)
+        if (savedInstanceState != null) {
             layoutManagerSavedState = savedInstanceState.getParcelable(LAYOUT_MANAGER_STATE);
+            prevLeftAction  = savedInstanceState.getString("prevLeftAction", "");
```

Comment:
Do we need this here? I think `onResume()` is called when the activity is restored. Therefore the settings will be reloaded. So I think we can remove the logic from both the `onViewStateRestored()` and `onSaveInstanceState()` right? Or am I missing something here?
```suggestion
```

Reply:
The reason I saved and restored it was that in `onResume`, the class may have been destroyed, but now that I think about it, in that case the drawables need to be reloaded anyway, so you are right, it's not needed.